File: Classroom/04.03/YetAnotherDividing.py
# https://codeforces.com/problemset/problem/1249/A

cases = int(input())

for i in range(cases):
    qty = int(input())

    students = map(int, input().split())

    flag =  False

    students = list(students)

    for j in range(qty):
        if flag:
            print("2")
            break
        for k in range(j, qty):
            if abs(students[j] - students[k]) == 1:
                flag = True
                break
    
    if not flag:
        print("1")

File: Classroom/04.03/YetAnotherDividing.cpp
// https://codeforces.com/problemset/problem/1249/A

#include <bits/stdc++.h>

using namespace std;

#define vi vector<int>
#define pb push_back

int main(){
    
    int cases;

    cin >> cases;

    for(int i = 0 ; i < cases ; i++){
        int qty;
        bool flag = false;
        vi students;

        cin >> qty;

        for(int j = 0 ; j < qty ; j++){
            int aux;
            cin >> aux;

            students.pb(aux);
        }

        for(int j = 0 ; j < qty and !flag ; j++){
            for(int k = j ; k < qty ; k++){
                if(abs(students[j] - students[k]) == 1){
                    flag = true;
                    break;
                }
            }
        }
        flag ? cout << "2" << "\n" : cout << "1" << "\n";
    }
    
    return 0;
}